### PR TITLE
Enhancement/update zxp on new version

### PR DESCRIPTION
**Pull request by BenSouchet**
To check the feature:
### Requirements

1. Install CMake (https://cmake.org/download/ Windows x64 Installer)
2. Install Visual Studio (enble Python develpment and Desktop develpment wih C++) (https://visualstudio.microsoft.com/fr/downloads/ Community edition)
3. Install Inno Setup (https://jrsoftware.org/isinfo.php)
4. **Very important**: Add Inno Setup (`C:\Program Files (x86)\Inno Setup 6`) to your environment PATH
5. 
Screenshot 2024-08-23 163710

### Steps
1. Ensure your OpenPype local folder is properly configured and is up to date with all the code you want to include in the release
2. Navigate with your terminal to your OpenPype folder, then build OpenPype with the command: `.\tools\build.ps1`
3. Generate the installer with: `.\tools\build_win_installer.ps1` (The installer will be placed in the `.\build\` folder)
6. **VERY VERRY IMPORTANT**: Install it but replace the destination path to point to `3.16.7`

## Then please clear your app data version folder

```
C:\Users\{YOUR_USERNAME}\AppData\Local\pypeclub\openpype\3.16
```

## You can now test OP

### CODE REVIEW

File: igniter/update_thread.py [hunk 1]
```diff
@@ -39,6 +47,73 @@ def _set_result(self, value):
...
+        for host in hosts:
+            extension_id, extension_version = self._extract_zxp_info_from_manifest(version_path, host)
+
+            if not extension_id or not extension_version:
+                # ZXP extension seems invalid, skipping
+                continue
+
+            # Remove installed ZXP extension
+            self.step_text_signal.emit("Removing installed ZXP extension for <b>{}</b> ...".format(host))
+            subprocess.run([str(path_prog), "/remove", extension_id])
```
Comment (BenSouchet) [inline igniter/update_thread.py:99]: on macos it's `--remove`

File: igniter/update_thread.py [hunk 2]
```diff
@@ -39,6 +47,73 @@ def _set_result(self, value):
...
+
+            # Install ZXP shipped in the current version folder
+            fullpath_curr_zxp_extension = version_path.joinpath("openpype", "hosts", host, "api", "extension.zxp")
+            if not fullpath_curr_zxp_extension.exists():
+                self.log_signal.emit("Cannot find ZXP extension for {}, looked at: {}".format(
+                    host, str(fullpath_curr_zxp_extension)), True)
+                continue
+
+            self.step_text_signal.emit("Install ZXP extension for <b>{}</b> ...".format(host))
+            completed_process = subprocess.run([str(path_prog), "/install", str(fullpath_curr_zxp_extension)],
```
Comment (BenSouchet) [inline igniter/update_thread.py:109]: on macos it's `--install`

Comment (BenSouchet) [reply igniter/update_thread.py]: Fixed

File: igniter/update_thread.py (hunk 1 above)
Comment (BenSouchet) [reply igniter/update_thread.py]: Fixed

**Comment by ccaillot** [comment]
All good !

**Pull request merged**